test/unit/gateways/d_local_test.rb

```diff
@@ -38,9 +38,10 @@ def test_successful_purchase
 
   def test_purchase_with_save
     stub_comms do
-      @gateway.purchase(@amount, @credit_card, @options.merge(save: true))
-    end.check_request do |_endpoint, data, _headers|
+      @gateway.purchase(@amount, @credit_card, @options.merge({ save: true, application_id: 'ActiveMerchant' }))
```

Comment:
`application_id` does not get passed in via options. It is a class variable so you should set it like `ActiveMerchant::Billing::DLocalGateway.application_id = 'ActiveMerchant'`